lib/index.js

```diff
@@ -447,61 +447,85 @@ ABI.stringify = function (types, values) {
   return ret
 }
 
+ABI.solidityHexValue = function (type, value, bitsize) {
+  // pass in bitsize = null if use default bitsize
+  var size, num
+  if (isArray(type)) {
+    var subType = type.replace(/\[.*?\]/, '')
+    if (!isArray(subType)) {
+      var arraySize = parseTypeArray(type)
+      if (arraySize !== 'dynamic' && arraySize !== 0 && value.length > arraySize) {
+        throw new Error('Elements exceed array size: ' + arraySize)
+      }
+    }
+    var arrayValues = value.map(function (v) {
+      return ABI.solidityHexValue(subType, v, 256)
+    })
+    return Buffer.concat(arrayValues)
+  } else if (type === 'bytes') {
```

Comment:
Ok, did also some line-by-line review of the `isArray` part here and some manual tests. Since this is already some reviewed write-up of an existing PR and solid new tests are added and passing I think I can give this a go. Thanks @djrtwo for submitting and @alex-forshtat-tbk for putting this along the finish line. 😄